### DISCUSSION TITLE
Help in hubert training from scratch

### DISCUSSION THREAD

**Discussion by pratikkumar018** [espnet/espnet#4454] [Q&A]
I am trying to train Hubert from scratch using this recipe https://github.com/espnet/espnet/tree/master/egs2/TEMPLATE/ssl1.

After iter0 is training, the mask acc saturates around 0.5 with unmask acc going till 0.8. 

When I am trying to run iter1 , in stage 5 the K Means step , it is generating HuBERT6_km500_token_list_iter1 folder in data with dict.txt & token.txt files.  When I open these files it only contains around 70-80 values instead of 500. I am passing n_clusters_iter1=500 but despite that it is generating only 70-80 values. Somehow my iter0 model is not able to generate 500 clusters with the dataset. I tried generating clusters with their hubert_base model and I was able to get 500 clusters. It looks like my iter0 model is not training rightly or I am missing something. Can you please suggest what may go wrong & what else I can try.

**Comment by sw005320** [comment]
Can you paste the learning curve for iter0?
Also, I added @simpleoier for this discussion.

**Comment by pratikkumar018** [comment]
These are the learning curves for mask & unmask acc & loss
![image](https://user-images.githubusercontent.com/80521981/174114200-d5b8bdcc-bfb1-45ec-b798-518a07043fc4.png)

![image](https://user-images.githubusercontent.com/80521981/174114284-2a72d695-ab32-4eb0-96b2-84cd13623c2e.png)
![image](https://user-images.githubusercontent.com/80521981/174114396-ed1b0054-43d4-4a10-b7a6-eaa63dadbe80.png)
